core/blockchain.go

```diff
@@ -1204,120 +1183,116 @@ func (bc *BlockChain) InsertReceiptChain(blockChain types.Blocks, receiptChain [
 		}
 		return false
 	}
+
```

Comment:
This PR makes significant changes to `InsertReceiptChain` but it is also missing a test that was included in the upstream PR `TestInsertReceiptChainRollback`. Looking at the results of coverage I can see that none of the error cases of `insertReceiptChain` are executed so I think it would be good to include a test here that exercises some of the error cases for `InsertReceiptChain`.

Reply:
The new test uses sidechains, which is why we can't run it. But the old test uses a test field/test mechanism "testInsert" to provoke the tests to be tested, which was removed in this refactor. The errors are tested using sidechains, which we don't have. 

Re-adding the mechanism to make it testeable was probably a change that should be made in a different PR for the same reason you mentioned the redundant parameter one

Reply:
https://github.com/celo-org/celo-blockchain/issues/1917